airflow/www/templates/airflow/trigger.html

```diff
@@ -47,22 +129,90 @@ <h2>Trigger DAG: {{ dag_id }}</h2>
         <input type="text" class="form-control" placeholder="Run ID" name="run_id">
       </div>
     </div>
+    {% if recent_confs|length > 0 %}
     <div class="form-group row">
       <div class="col-md-2">
-        <label for="recent_configs">Recent Configurations</label>
+        <label for="recent_configs">Select Recent Configurations</label>
         <select class="form-control" name="recent_configs" id="recent_configs">
-          <option value="">empty config</option>
-          {% for conf in recent_confs %}
-          <option value="{{ conf }}">{{ conf }}</option>
+          <option value="{{ conf }}">Default parameters</option>
+          {% for run_id, recent_conf in recent_confs.items() %}
+          <option value="{{ recent_conf }}">{{ run_id }}: {{ recent_conf }}</option>
           {% endfor %}
         </select>
       </div>
     </div>
-    <label for="conf">Configuration JSON (Optional, must be a dict object)</label>
-    <textarea class="form-control" name="conf" id="json">{{ conf }}</textarea>
+    {% endif %}
+    {%- if form_fields %}
+    <div class="panel panel-primary ">
+      <div class="panel-heading">
+        <h4 class="panel-title">DAG conf Parameters</h4>
+      </div>
+      <textarea style="display: none;" id="json_start" name="json_start">
+        {{- conf -}}
+      </textarea>
+      <textarea style="display: none;" id="form_sections" name="form_sections">
+        {{- form_fields.values() | map(attribute="schema.section", default="") | reject("equalto", "") | unique() | join(",") | lower() | replace(" ", "_") -}}
+      </textarea>
+      <div class="panel-body">
 
-    <br />
+        <table class="table table-hover table-bordered">
+          <tbody>
+            {% for form_key, form_details in form_fields.items() %}
+            {% if ("section" not in form_details.schema or not form_details.schema.section) and ("const" not in form_details.schema or not form_details.schema.const) %}
+            {{ form_element(form_key, form_details) }}
+            {% endif %}
+            {% endfor %}
+          </tbody>
+        </table>
 
+        {% for form_section, form_items in form_fields.values() | groupby(attribute="schema.section", default="") %}
+        {% if form_section %}
+        <div class="panel-group" id="accordion_section_{{ form_section | lower() | replace(' ', '_') }}">
+          <div class="panel panel-default">
+            <div class="panel-heading">
+              <h4 class="panel-title">
+                <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion_section_{{ form_section | lower() | replace(' ', '_') }}" href="#accordion_section_{{ form_section | lower() | replace(' ', '_') }}_href"
+                  data-original-title="" title="" id="{{ form_section | lower() | replace(' ', '_') }}_toggle">{{ form_section }}<span class="caret"></span></a>
+              </h4>
+            </div>
```

Comment:
If the form defines multiple sections, this generate a new section header - each section is folded by default and can be un-folded:
![image](https://user-images.githubusercontent.com/95105677/214952183-4a9f23bc-4b71-409d-8888-4001cf6d93b3.png)